core/src/test/java/org/elasticsearch/action/support/master/IndexingMasterFailoverIT.java

```diff
@@ -73,9 +73,10 @@ public void testMasterFailoverDuringIndexingWithMappingChanges() throws Throwabl
                 .put("discovery.zen.join_timeout", "10s")  // still long to induce failures but to long so test won't time out
                 .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "1s") // <-- for hitting simulated network failures quickly
                 .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 2)
+                .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0")  // <-- we wait for cluster formation at the end
```

Comment:
answering out of order:

> which is completely unnecessary (I've even outlined a solution)

My goal here is to remove the complexity and asynchronicity of node starting. Adding that back will defeat what I wanted to achieve. I went through the tests and didn't see any test that actually relies on this async nature (except for bypassing this waiting).

> Calling startNodes(3) should not trip up because of some internals require an extra setting to be passed.

To be clear - if you use that method when `min_master_nodes` is auto managed you don't need to do anything. It's only if you "own" that setting than you have to also deal with the sequential nature of starting (unless you use this setting which IMO a valid use case and the reason for it's existence).

> It exposes the sequential nature that is used by startNodes to users of this API,

I can document that and make it official? I very much prefer that to doing async work in the test cluster.